backends/etcdv3/client.go

```diff
@@ -11,16 +11,22 @@ import (
 	"golang.org/x/net/context"
 
 	"github.com/coreos/etcd/clientv3"
+	"sync"
 )
 
 // Client is a wrapper around the etcd client
 type Client struct {
-	client *clientv3.Client
+	client  *clientv3.Client
+	watches map[string]clientv3.WatchChan
+	mu      sync.Mutex
```

Comment:
The `WatchPrefix` is called synchronously (https://github.com/kelseyhightower/confd/blob/20b3d37da7aaa2c176c0612202c06c5ba4f7d987/resource/template/processor.go#L96), so there is no need to create a mutex. Remove it.